File: src/main/java/ir/sayandevelopment/listener/PlayerMentionListener.java
package ir.sayandevelopment.listener;

import ir.sayandevelopment.spigot.SpigotMain;
import me.sayandevelopment.sayanchat.api.event.PlayerMentionEvent;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;

public class PlayerMentionListener implements Listener {

    @EventHandler
    public void onMention(PlayerMentionEvent event) {
        Player target = event.getMentioned();
        if (SpigotMain.vanishedPlayers.containsKey(target.getUniqueId()) && SpigotMain.vanishedPlayers.get(target.getUniqueId()).isVanished()) {
            event.setCancelled(true);
        }
    }
}

File: src/main/java/ir/syrent/velocityvanish/spigot/listener/PlayerMentionListener.kt
package ir.syrent.velocityvanish.spigot.listener

import ir.syrent.velocityvanish.spigot.VelocityVanishSpigot
import ir.syrent.velocityvanish.spigot.ruom.Ruom
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener

class PlayerMentionListener(
    private val plugin: VelocityVanishSpigot
) : Listener {

    init {
        Ruom.registerListener(this)
    }

}